WalletWasabi/WabiSabi/Models/MultipartyTransaction/Signing.cs

```diff
@@ -0,0 +1,88 @@
+using NBitcoin;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using WalletWasabi.WabiSabi.Backend.Models;
+
+namespace WalletWasabi.WabiSabi.Models.MultipartyTransaction
+{
+	public record Signing(Parameters Parameters, ImmutableArray<Coin> Inputs, ImmutableArray<TxOut> Outputs) : State
+	{
+		public ImmutableDictionary<int, WitScript> Witnesses { get; init; } = ImmutableDictionary<int, WitScript>.Empty;
+
+		public bool IsFullySigned => Witnesses.Count() == Inputs.Length;
+
+		public FeeRate EffectiveFeeRate => new FeeRate(Balance, EstimatedVsize);
+
+		public Money Balance => Inputs.Sum(x => x.Amount) - Outputs.Sum(x => x.Value);
+
+		public int EstimatedVsize => Parameters.SharedOverhead + EstimatedInputsVsize + OutputsVsize;
+		public int EstimatedInputsVsize => Inputs.Sum(x => x.TxOut.ScriptPubKey.EstimateInputVsize());
+		public int OutputsVsize => Outputs.Sum(x => x.ScriptPubKey.EstimateOutputVsize());
+
+		public IEnumerable<Coin> UnsignedInputs => Enumerable.Range(0, Inputs.Length).Where(i => !IsInputSigned(i)).Select(i => Inputs[i]);
```

Comment:
Untested but it should work.

```suggestion
		public IEnumerable<Coin> UnsignedInputs => Inputs.Where((_, i) => !IsInputSigned(i));
```